### PR TITLE
Add hash_algo parameter to find_latest_working_version

### PR DESCRIPTION
**Pull request by greyblue9**

### CODE REVIEW

File: src/use/use.py [hunk 1]
```diff
@@ -455,7 +456,8 @@ def find_latest_working_version(releases: Dict[str, List[Dict[str, str]]], # {ve
             if is_version_satisfied(info, sys_version) and \
                 is_platform_satisfied(info, platform_tags) and \
                 is_interpreter_satisfied(info, interpreter_tag):
-                return info["version"], info["hash"]
+                hash_value = info["digests"].get(hash_algo.name)
```
Comment (amogorkon) [inline src/use/use.py:459]: might need to reconsider this step and raise an exception instead if no hash is available

File: src/use/use.py [hunk 2]
```diff
@@ -928,7 +930,9 @@ def _use_str(
                 else:
                     try:
                         data = response.json()
-                        hit = find_latest_working_version(data["releases"])
+                        hit = find_latest_working_version(data["releases"], hash_algo=hash_algo)
```
Comment (amogorkon) [inline src/use/use.py:933]: hash_algo is an enum member, it should be hash_algo.name to get the str
```suggestion
                        hit = find_latest_working_version(data["releases"], hash_algo=hash_algo.name)
```

File: tests/unit_test.py [hunk 3]
```diff
@@ -152,15 +152,17 @@ def test_auto_install_native():
     except RuntimeWarning as w:
       rw = w
     assert rw, "Expected a RuntimeWarning from unversioned auto-install"
-    params = re.search(
+    match:Optional[re.Match] = re.search(
```
Comment (amogorkon) [inline tests/unit_test.py:155]: heh. funky annotation :)

File: tests/unit_test.py [hunk 4]
```diff
@@ -152,15 +152,17 @@ def test_auto_install_native():
...
       "use\\("
         "\"(?P<name>.*)\", "
         "version=\"(?P<version>.*)\", "
         "hash_value=\"(?P<hash_value>.*)\", "
         "auto_install=True"
       "\\)",
       rw.args[0]
-    ).groupdict()
```
Comment (amogorkon) [inline tests/unit_test.py:163]: does that work now?

**Review by amogorkon — CHANGES_REQUESTED** [review]

File: tests/unit_test.py (hunk 3 above)
Comment (greyblue9) [reply tests/unit_test.py:155]: Thank you! 🔥

File: tests/unit_test.py (hunk 4 above)
Comment (greyblue9) [reply tests/unit_test.py:163]: Yes, it does, it's jappy with the `version, hash_value = hit` added elsewhere in the main file

File: src/use/use.py [hunk 5]
```diff
@@ -928,7 +931,12 @@ def _use_str(
                 else:
                     try:
                         data = response.json()
-                        hit = find_latest_working_version(data["releases"])
+                        hit = find_latest_working_version(
+                            data["releases"],
+                            hash_algo=hash_algo.name,
+                        )
+                        if hit:
+                          version, hash_value = hit
```
Comment (greyblue9) [inline src/use/use.py:939]: This fixes the issue found by the unit test (in later question).

File: src/use/use.py (hunk 2 above)
Comment (greyblue9) [reply src/use/use.py]: Addressed in commit #2

**Review by amogorkon — APPROVED** [review]

File: src/use/use.py (hunk 1 above)
Comment (greyblue9) [reply src/use/use.py]: Addressed in commit #3

**Pull request merged**